content/running_bookwyrm/external-storage.md

```diff
@@ -0,0 +1,120 @@
+- `AWS_S3_REGION_NAME`: e.g. `"eu-west-1"` for AWS, `"fr-par"` for Scaleway or `"nyc3"` for Digital Ocean 
+
+If your S3-compatible service is Amazon AWS, you should be set. If not, you’ll have to uncomment the following lines:
+
+- `AWS_S3_CUSTOM_DOMAIN`: the domain that will serve the assets, e.g. `"example-bucket-name.s3.fr-par.scw.cloud"` or `"${AWS_STORAGE_BUCKET_NAME}.${AWS_S3_REGION_NAME}.digitaloceanspaces.com"`
+- `AWS_S3_ENDPOINT_URL`: the S3 API endpoint, e.g. `"https://s3.fr-par.scw.cloud"` or `"https://${AWS_S3_REGION_NAME}.digitaloceanspaces.com"`
+
+### Copying local media to external storage
+
+If your BookWyrm instance is already running, and media have been uploaded (user avatars, book covers…), you will need to migrate uploaded media to your bucket.
+
+This task is done with the command: 
+
+```bash
+./bw-dev copy_media_to_s3
+```
+
+### Enabling external storage for BookWyrm
+
+To enable the S3-compatible external storage, you will have to edit your `.env` file by changing the property value for `USE_S3` from `False` to `True`:
+
+```
+USE_S3=True
+```
+
+#### Static assets
+
+Then, you will need to run the following command, to copy the static assets to your S3 bucket:
+
+```bash
+./bw-dev collectstatic
+```
+
+#### CORS settings
+
+Once the static assets are collected, you will need to set up CORS for your bucket.
+
+Some services like Digital Ocean provide an interface to set it up, see [Digital Ocean doc: How to Configure CORS](https://docs.digitalocean.com/products/spaces/how-to/configure-cors/).
+
+If your service doesn’t provide an interface, you can still set up CORS with the command line.
+
+Create a file called `cors.json`, with the following content:
+
+```json
+{
+  "CORSRules": [
+    {
+      "AllowedOrigins": ["http://MY_DOMAIN_NAME", "http://www.MY_DOMAIN_NAME"],
+      "AllowedHeaders": ["*"],
+      "AllowedMethods": ["GET", "HEAD", "POST", "PUT", "DELETE"],
+      "MaxAgeSeconds": 3000,
+      "ExposeHeaders": ["Etag"]
+    }
+  ]
+}
+```
+
+Replace `MY_DOMAIN_NAME` with the domain name(s) of your instance.
+
+Then, run the following command:
+
+```bash
+./bw-dev set_cors_to_s3 cors.json
+```
+
+No output means it should be good.
+
+If you are starting a new BookWyrm instance, you can go back to the setup instructions right now. If not, keep on reading.
+
+### Restarting your instance
+
+Once the media migration has been done and the static assets are collected, you can restart your instance with:
+
+```bash
+docker compose restart
+```
+
+If all goes well, your storage has been changed without server downtime. If some fonts are missing (and your browser’s JS console lights up with alerts about CORS), something went wrong [here](#cors-settings). In that case it might be good to check the headers of a HTTP request against a file on your bucket:
+
+```bash
+curl -X OPTIONS -H 'Origin: http://MY_DOMAIN_NAME' http://BUCKET_URL/static/images/logo-small.png -H "Access-Control-Request-Method: GET"
```

Comment:
This seems like a useful test, does it make sense to add it as a step before restarting the instance? For folks who aren't familiar with what curl does, it would be helpful to clarify what success and failure look like.